pyomo/contrib/community_detection/detection.py

```diff
@@ -168,7 +170,225 @@ def detect_communities(model, type_of_community_map='constraint', with_objective
 
     # Return community_map, which has integer keys that now map to a tuple of two lists
     # containing Pyomo modeling components
-    return community_map
+    return CommunityMap(community_map, model, type_of_community_map, with_objective, weighted_graph, random_seed)
+
+
+class CommunityMap(object):
+
+    def __init__(self, community_map, model, type_of_community_map, with_objective, weighted_graph, random_seed):
+        self.community_map = community_map
+        self.model = model
+        self.type_of_community_map = type_of_community_map
+        self.with_objective = with_objective
+        self.weighted_graph = weighted_graph
+        self.random_seed = random_seed
+
+    def __repr__(self):
+        return str(self.stringify())
+
+    def __iter__(self):
+        for key in self.community_map:
+            yield key
+
+    def __getitem__(self, item):
+        return self.community_map[item]
+
+    def keys(self):
+        return self.community_map.keys()
+
+    def values(self):
+        return self.community_map.values()
+
+    def items(self):
+        return self.community_map.items()
+
+    def draw_graph(self, type_of_graph='constraint', with_objective=True, weighted_graph=True, random_seed=None, pos=None):
```

Comment:
Why have all of these parameters when `self` should have the information?

Reply:
In the case of the draw_graph function, the CommunityMap object only tells us how to color the nodes according to their communities. But we can control whether we draw a constraint-node graph, a variable-node graph, or a bipartite graph (this is what the 'type_of_graph' parameter is for). The other parameters are similarly involved in the specifics of the graph that we draw (whether we treat the objective as a node and whether we want to provide a layout with 'pos').

I forgot to remove 'random_seed' as a parameter (this is used for creating community maps), so maybe this led to some confusion as well - sorry about that!